src/business_logic/state/cached_state.rs

```diff
@@ -80,10 +82,14 @@ impl<T: StateReader + Clone> StateReader for CachedState<T> {
 
     fn get_class_hash_at(&mut self, contract_address: &Address) -> Result<&[u8; 32], StateError> {
         if self.cache.get_class_hash(contract_address).is_none() {
-            let class_hash = self.state_reader.get_class_hash_at(contract_address)?;
+            let class_hash = match self.state_reader.get_class_hash_at(contract_address) {
+                Ok(x) => Cow::Borrowed(x),
+                Err(StateError::NoneContractState(_)) => Cow::Owned([0; 32]),
```

Comment:
Just a question.
Are you sure that this get should return [0; 32] in this cases

Reply:
No, however that's what is actually inserted in `class_hash_initial_values`.